From e3cd17dc92e649480c5a7184c352c2dd6434382a Mon Sep 17 00:00:00 2001
From: Dongjin Kim <tobetter@gmail.com>
Date: Fri, 11 Feb 2022 05:08:35 +0900
Subject: [PATCH 88/99] ODROID-N2/C4: arm64/dts: add PPS client device trees

Two device tree overlays are added to support GPS PPS client through
GPIO pin header #11 and #12 as 'pps-gpio_p11.dtbo' and
'pps-gpio_p12.dtbo' respectively.

Signed-off-by: Dongjin Kim <tobetter@gmail.com>
Change-Id: I67c0a1750de0885e27b8545fab8acbd4ef0b7ef9
---
 .../dts/amlogic/overlays/odroidc4/Makefile    |  2 ++
 .../overlays/odroidc4/pps-gpio_p11.dts        | 19 +++++++++++++++++++
 .../overlays/odroidc4/pps-gpio_p12.dts        | 19 +++++++++++++++++++
 .../dts/amlogic/overlays/odroidn2/Makefile    |  2 ++
 .../overlays/odroidn2/pps-gpio_p11.dts        | 19 +++++++++++++++++++
 .../overlays/odroidn2/pps-gpio_p12.dts        | 19 +++++++++++++++++++
 6 files changed, 80 insertions(+)
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p11.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p12.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p11.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p12.dts

diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile
index 289a2bbda1a4..54ee8bc8bcc6 100644
--- a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/Makefile
@@ -5,6 +5,8 @@ dtbo-y += \
 	i2c0.dtbo \
 	i2c1.dtbo \
 	pcf8563.dtbo \
+	pps-gpio_p11.dtbo \
+	pps-gpio_p12.dtbo \
 	spi0.dtbo \
 	sx865x-i2c1.dtbo \
 	uart0.dtbo \
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p11.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p11.dts
new file mode 100644
index 000000000000..85e107cc28d8
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p11.dts
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/meson-g12a-gpio.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	fragment@0 {
+		target-path = "/";
+
+		__overlay__ {
+			pps: pps {
+				compatible = "pps-gpio";
+				gpios = <&gpio GPIOX_3 GPIO_ACTIVE_HIGH>;
+				status = "okay";
+			};
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p12.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p12.dts
new file mode 100644
index 000000000000..fa02e68ca776
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidc4/pps-gpio_p12.dts
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/meson-g12a-gpio.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	fragment@0 {
+		target-path = "/";
+
+		__overlay__ {
+			pps: pps {
+				compatible = "pps-gpio";
+				gpios = <&gpio GPIOX_16 GPIO_ACTIVE_HIGH>;
+				status = "okay";
+			};
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidn2/Makefile b/arch/arm64/boot/dts/amlogic/overlays/odroidn2/Makefile
index a1482759fa95..6467074b7896 100644
--- a/arch/arm64/boot/dts/amlogic/overlays/odroidn2/Makefile
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidn2/Makefile
@@ -3,6 +3,8 @@ dtbo-y += \
 	hktft35.dtbo \
 	i2c0.dtbo \
 	i2c1.dtbo \
+	pps-gpio_p11.dtbo \
+	pps-gpio_p12.dtbo \
 	pwm_cd.dtbo \
 	pwm_ef.dtbo \
 	pwmfan-lowtemp.dtbo \
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p11.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p11.dts
new file mode 100644
index 000000000000..85e107cc28d8
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p11.dts
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/meson-g12a-gpio.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	fragment@0 {
+		target-path = "/";
+
+		__overlay__ {
+			pps: pps {
+				compatible = "pps-gpio";
+				gpios = <&gpio GPIOX_3 GPIO_ACTIVE_HIGH>;
+				status = "okay";
+			};
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p12.dts b/arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p12.dts
new file mode 100644
index 000000000000..fa02e68ca776
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/odroidn2/pps-gpio_p12.dts
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/meson-g12a-gpio.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	fragment@0 {
+		target-path = "/";
+
+		__overlay__ {
+			pps: pps {
+				compatible = "pps-gpio";
+				gpios = <&gpio GPIOX_16 GPIO_ACTIVE_HIGH>;
+				status = "okay";
+			};
+		};
+	};
+};
-- 
2.37.3

